### ISSUE TITLE
ViTextEdit does neither accept #text nor allow writing via #object

### ISSUE THREAD

**Issue by codeZeilen** [hpi-swa/vivide#45]
I've tried to use the ViTextEdit to create a source code editor. First I've tried the following script:

```Smalltalk
[:in :out | ([:all | all collect: [:o |
    ([:obj | {
        [:thing | 
	    #text -> (thing getString) asText
	            <- [:text | self halt] ] value: obj }]
	value: o)]]
value: in) do: [:result | out add: result]]
-> {
	#isProperty -> true.
	#label -> 'TextEdit Demo'.
	#priority -> 60.
	#view -> ViTextEdit.
}
````

This leads to an empty ViTextEdit pane. I have then changed the code to use the object property instead. And removed the writing to #object.

```Smalltalk
[:in :out | ([:all | all collect: [:o |
    ([:obj | {
        [:thing | 
	    #object -> (thing getString) asText ] value: obj }]
	value: o)]]
value: in) do: [:result | out add: result]]
...
````

This leads to a box were the source code is displayed and can be edited, but not saved. This kind of makes sense, as the object of a TextEdit should be a text. I then readded the writing part:

```Smalltalk
[:in :out | ([:all | all collect: [:o |
    ([:obj | {
        [:thing | 
	    #object -> (thing getString) asText
	            <- [:text | self halt] ] value: obj }]
	value: o)]]
value: in) do: [:result | out add: result]]
...
````

The output in the pane is then:

```Smalltalk
[closure] in ViScriptPart class>>convertProperty:
````

This looks inconsistent to me. First I have expected it to use #text but I understand the reasoning to use #object as explained above. However, then I think the writing is probably broken.

The current way to solve this (according to the Vivide Package Browser) is to use a ViTextView and set the script property editable to true.

**Comment by marceltaeumel** [comment]
`ViTextEdit` modifies an actual text object. If you accept the changes, a modified version of the object will be send to the next pane.

For regular code editors, I would suggest `ViTextView`, which can be editable:

```Smalltalk
[:in :out | ([:all | all collect: [:o | (
	[:object | { 
		#text -> object contents
			<- [:n | object contents: n]}]
		value: o), { #object -> o }]]
			value: in) do: [:result | out add: result]]
-> {
	#isProperty -> true.
	#editable -> true.
	#view -> ViTextView.
}
```

Open this script, for example, on a `StringHolder`.

**Comment by codeZeilen** [comment]
Yes that's what I've found out and mentioned in the issue. My problem with this is that the names are misleading. My first reaction was to use a ViTextView to merely view text and when I was ready to add the edit functionality I switched to the ViTextEdit and did not get what I expected :D. Maybe at least add a class comment making this distinction clear?
